src/Test/Integration/Model/Payment/Magento/InvoiceTest.php

```diff
@@ -0,0 +1,145 @@
+ */
+
+namespace Shopgate\Import\Test\Integration\Model\Payment\Shopgate;
+
+use Exception;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Sales\Model\Order as MagentoOrder;
+use Magento\Sales\Model\OrderRepository;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\ObjectManager;
+use PHPUnit\Framework\TestCase;
+use Shopgate\Base\Tests\Integration\SgDataManager;
+use Shopgate\Import\Model\Service\Import;
+use ShopgateLibraryException;
+use ShopgateOrder;
+
+/**
+ * @magentoAppIsolation enabled
+ * @magentoDbIsolation  enabled
+ * @magentoAppArea      frontend
+ */
+class InvoiceTest extends TestCase
+{
+    /** @var ObjectManager $objectManager */
+    private $objectManager;
+    /** @var Import */
+    private $importClass;
+    /** @var OrderRepository */
+    private $orderRepository;
+    /** @var SgDataManager */
+    private $dataManager;
+
+    /**
+     * Integration test preparation
+     */
+    public function setUp()
+    {
+        $this->objectManager   = Bootstrap::getObjectManager();
+        $this->importClass     = $this->objectManager->create(Import::class);
+        $this->orderRepository = $this->objectManager->create(OrderRepository::class);
+        $this->dataManager     = $this->objectManager->create(SgDataManager::class);
+    }
+
+    /**
+     * @magentoConfigFixture current_store payment/checkmo/active 1
+     *
+     * @throws Exception
+     * @throws ShopgateLibraryException
+     * @throws InputException
+     * @throws NoSuchEntityException
+     *
+     * @dataProvider         paidFlagProvider
+     */
+    public function testPaymentMappingOnImport($isPaid): void
+    {
+        $result = $this->importClass->addOrder($this->getShopgateOrder($isPaid));
+        /** @var MagentoOrder $magentoOrder */
+        $magentoOrder = $this->orderRepository->get($result['external_order_id']);
+
+        $this->assertEquals('checkmo', $magentoOrder->getPayment()->getMethod());
+        $this->assertEquals('pending', $magentoOrder->getStatus());
+    }
+
+    /**
+     * @magentoConfigFixture current_store payment/checkmo/active 0
+     *
+     * @throws Exception
+     * @throws ShopgateLibraryException
+     * @throws InputException
+     * @throws NoSuchEntityException
+     */
+    public function testInactivePaymentMappingOnImport(): void
+    {
+        $result = $this->importClass->addOrder($this->getShopgateOrder());
+        /** @var MagentoOrder $magentoOrder */
+        $magentoOrder = $this->orderRepository->get($result['external_order_id']);
+
+        $this->assertEquals('shopgate', $magentoOrder->getPayment()->getMethod());
+        $this->assertEquals('pending', $magentoOrder->getStatus());
+    }
+
+
+    /**
+     * @param int $isPaid
+     *
+     * @return ShopgateOrder
+     *
+     * @throws Exception
+     */
+    private function getShopgateOrder(int $isPaid = 0): ShopgateOrder
```

Comment:
there seems to be some repetition happening here, maybe extend some sort of base class that has these helper methods